src/sift.jl

```diff
@@ -56,19 +56,21 @@ function iterate(iter::SiftIterable, state::SiftState)
     @debug "The number of maxima: $maxlen"
     state.fix_steps == iter.stop_steps && return nothing
     zerocrossing!(state.yvec, state.crosses)
-    abs(length(state.crosses) - maxlen - minlen) <=1 && (state.fix_steps +=1)
-    if maxlen<4 || minlen<4
+    abs(length(state.crosses) - maxlen - minlen) <= 1 && (state.fix_steps += 1)
+    if maxlen < 4 || minlen < 4
         return nothing
     end
     smin = get_edgepoint(state.yvec, state.xvec, state.mins, first, isless)
     smax = get_edgepoint(state.yvec, state.xvec, state.maxes, first, !isless)
     emin = get_edgepoint(state.yvec, state.xvec, state.mins, last, isless)
     emax = get_edgepoint(state.yvec, state.xvec, state.maxes, last, !isless)
 
-    maxTS = interpolate([first(state.xvec); state.xvec[state.maxes]; last(state.xvec)],[smax; state.yvec[state.maxes]; emax],state.xvec,DierckXInterp())
-    minTS = interpolate([first(state.xvec); state.xvec[state.mins]; last(state.xvec)],[smin; state.yvec[state.mins]; emin],state.xvec,DierckXInterp())
-    subs = (maxTS+minTS)/2
-    state.s = sum(abs,subs)
+    maxTS = interpolate([first(state.xvec); state.xvec[state.maxes]; last(state.xvec)],
+        [smax; state.yvec[state.maxes]; emax], state.xvec, DierckXInterp())
+    minTS = interpolate([first(state.xvec); state.xvec[state.mins]; last(state.xvec)],
```

Comment:
This is the same as above, I have to refactor this a bit to reduce the line length, but simply splitting makes it even harder to read.